client-app/shared/catalog/composables/useConfigurableProduct.ts

```diff
@@ -120,15 +142,19 @@ export function useConfigurableProduct(configurableProductId: string) {
     },
     {
       deep: true,
+      immediate: true,
     },
   );
 
   return {
     fetchProductConfiguration,
     selectSectionValue,
     loading: computed(() => fetching.value || creating.value),
-    configuration: readonly(configuration),
+    configuration: shallowReadonly(configuration),
```

Comment:
lets keep just readonly. Less questions. Or if needed - add setter to the composable